frontend/evaluations/views.py

```diff
@@ -3,114 +3,116 @@
+
+    # # * Check for invalid characters
+    # if not input_text_regex.match(input_text):
+    #     return JsonResponse({"error_msg": "Invalid input. The input contains invalid characters.", "status": 400})    
+
+    # * Check for authentication
+    user = request.user
+    if user.is_authenticated:
+        validated_text = requests.get(backend_url, params={"text" : input_text})
+
+        if validated_text.status_code == 503:
+            request_queue.put((request, input_text, backend_url))
+            return JsonResponse({"status" : 503})
+        
+        while not request_queue.empty():
+            # Pokud je fronta neprázdná, převezměte request z fronty
+            queued_request, queued_input_text, queued_backend_url = request_queue.get()
+            queued_validated_text = requests.get(queued_backend_url, params={"text": queued_input_text})
+            
+            if queued_validated_text.status_code == 503:
+                # Pokud je stále kód 503, přidejte request zpět na konec fronty
+                request_queue.put((queued_request, queued_input_text, queued_backend_url))
+                break
+            else:
+                # Pokud kód 503 není, zpracujte request
+                queued_request_json = eval_fnc(queued_request, queued_input_text, queued_backend_url)
+                if queued_request_json.status_code == 503:
+                    # Pokud byl request opět přidán do fronty, přidejte ho zpět na konec fronty
+                    request_queue.put((queued_request, queued_input_text, queued_backend_url))
+                    break
 
-# Create your views here.
-def evaluation_view(request, *args, **kwargs):
-    text = request.GET["text"]
-    validated_text = requests.get("http://127.0.0.1:8002/backend/v1/eval", params={"text" : text}).json()
-    
-    # //print(type(validated_text[0]["label"]))
-
-    # * New Evaluation Block creation
-    whole_claim = " ".join([claim["claim"] for claim in validated_text])
-    new_evaluation_block = Evaluation_block.objects.create(claims=whole_claim)
-
-    for evaluation in validated_text:
-        new_evaluation = Evaluation.objects.create(
-            evaluation_block=new_evaluation_block,
-
-            claim=evaluation.get("claim"),
-            label=evaluation.get("label"), 
-            supports=evaluation.get("supports"), 
-            refutes=evaluation.get("refutes"),
-            ei=evaluation.get("ei"),
-            nei=evaluation.get("nei"),
-            evidence=evaluation.get("evidence")
-        )
-        evaluation["id"] = new_evaluation.id # ! Adding id to the obtained JSON -> passing to feedbacks app
-        evaluation["evaluation_block"] = new_evaluation_block.id
 
-    return JsonResponse({"validated" : validated_text})
+        # * New Evaluation Block creation
+        whole_claim = " ".join([claim["claim"] for claim in validated_text.json()])
+        new_evaluation_block = Evaluation_block.objects.create(user=user, claims=whole_claim)
```

Comment:
Add some error handling

```suggestion
try:
        whole_claim = " ".join([claim["claim"] for claim in validated_text.json()])
        new_evaluation_block = Evaluation_block.objects.create(user=user, claims=whole_claim)
except:
        print("Error")
```